### PR TITLE
High priority feedback requests.

### PR DESCRIPTION
**Pull request by connor-young-dev**
- Swap activity colours

- Increase font sizes in categories for the titles. These are h1 titles that are getting the h2 class added. For consistency I have modified the renderer to use the h1 class.

- Change body and link colour for just copy. Had to create new variables and target region specifically as link and text changes had big changes in area's they wouldn't want.

- Reintroduced the breadcrumb. Don't show unless more then one item, unless in a course.

- Removed activities from the course index menu so only sections show. Changed up the template to remove dropdown functionality and entirely link to a section. (Not a great change).

- Applied quiz styling to match the mockups.

- TODO Question type styling. Made a start to the drag and drop text question type and fixed the bug with font sizing.

- Bumped version.

### CODE REVIEW

File: classes/boostnavbar.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
+<?php
```
Comment (gthomas2) [inline classes/boostnavbar.php:1]: We should put the regular moodle licence at the top + comment for class with copyright, author, etc...

File: classes/boostnavbar.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        $this->remove('courses');
+
+        // Remove the course breadcrumb node.
+        $this->remove($this->page->course->id, \breadcrumb_navigation_node::TYPE_COURSE);
+
+        // Remove the navbar nodes that already exist in the secondary navigation menu.
+        $this->remove_items_that_exist_in_navigation($PAGE->secondarynav);
+
+        switch ($this->page->pagetype) {
+            case 'group-groupings':
```
Comment (gthomas2) [inline classes/boostnavbar.php:86]: you don't need to have empty cases - just add the cases where you want to act.

File: classes/boostnavbar.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        }
+
+        $itemcount = $this->item_count();
+        if ($itemcount <= 0) {
+            return;
+        }
+
+        $this->items = array_values($this->items);
+        // Set the last item to last item if it is not.
+        $lastitem = $this->items[$itemcount - 1];
```
Comment (gthomas2) [inline classes/boostnavbar.php]: tip - $lastitem = end($this->items);

File: classes/output/core_renderer.php [hunk 4]
```diff
@@ -106,4 +116,67 @@ public function additional_html_footer() {
...
         return '';
     }
+
+    /**
+     * Renders the header bar.
+     *
+     * @param context_header $contextheader Header bar object.
+     * @return string HTML for the header bar.
+     */
+    protected function render_context_header(\context_header $contextheader) {
```
Comment (gthomas2) [inline classes/output/core_renderer.php]: you could add a return type - : string

File: scss/styles/coursecategory.scss [hunk 5]
```diff
@@ -1,4 +1,15 @@
...
     display: none;
+}
+
+// Remove Boost small styles to category names.
+.course_category_tree .category > .info > .categoryname {
+    @include font-size($h3-font-size);
+}
+
+// Increase size of coursenames.
+.coursebox>.info>.coursename {
```
Comment (gthomas2) [inline scss/styles/coursecategory.scss]: Add spaces between arrows

File: scss/styles/general.scss [hunk 6]
```diff
@@ -85,4 +85,12 @@ body {
 }
 .custom-switch .custom-control-input:checked~.custom-control-label::after {
     background-color: #fff;
+}
+
+// Link colour just for body copy.
+#page-content p {
+    color: $body-copy-color;
```
Comment (gthomas2) [inline scss/styles/general.scss:92]: You can simplify this I think

```
#page-content p {
    &, a {
        color: $body-copy-color;
    }
}

File: templates/core_courseformat/local/courseindex/section.mustache [hunk 7]
```diff
@@ -0,0 +1,37 @@
+<a
+        class="courseindex-section d-block {{#current}}current{{/current}}"
```
Comment (gthomas2) [inline templates/core_courseformat/local/courseindex/section.mustache]: Weird indentation here. Not 100% sure what the coding standards are for mustache templates but it feels like the indentation should be 4 spaces here.

File: templates/core_courseformat/local/courseindex/section.mustache [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        data-for="section"
+        data-id="{{id}}"
+        data-number="{{number}}"
+        role="treeitem"
+        aria-owns="courseindexsection{{number}}"
+
+>
+    <div class="courseindex-item d-flex
+            {{^visible}}dimmed{{/visible}}
+        {{#hasrestrictions}}restrictions{{/hasrestrictions}}
```
Comment (gthomas2) [inline templates/core_courseformat/local/courseindex/section.mustache:13]: indentation weird

File: templates/core_courseformat/local/courseindex/section.mustache [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+            courseindex-section-title"
+         data-for="section_item"
+    >
+        <span
+                class="courseindex-link text-truncate"
```
Comment (gthomas2) [inline templates/core_courseformat/local/courseindex/section.mustache]: indentation

File: templates/core_courseformat/local/courseindex/section.mustache [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        </span>
+        <span class="courseindex-locked ml-1" data-for="cm_name">
+            {{#pix}} t/locked, core {{/pix}}
+        </span>
+        <span class="dragicon ml-auto">{{#pix}}i/dragdrop{{/pix}}</span>
+    </div>
+</a>
+{{#js}}
+    require(['core_courseformat/local/courseindex/section'], function(component) {
+    component.init('{{uniqid}}-course-index-section-{{id}}');
```
Comment (gthomas2) [inline templates/core_courseformat/local/courseindex/section.mustache]: This line needs indenting by an extra 4 spaces

File: classes/boostnavbar.php (hunk 2 above)
Comment (connor-young-dev) [reply classes/boostnavbar.php:86]: Just shorthand, not empty.

**Review by gthomas2 — APPROVED** [review]

**Pull request merged**